### ISSUE TITLE
Invalid dashboard states generate 404 errors instead of metrics configs screen with actionable error messages

### ISSUE THREAD

**Issue by magorlick** [rilldata/rill-developer#1144]
**Describe the bug**
Seeing additional navigation related 404 errors user reports.
_"I wasn't able to use the model in the dashboard - kept getting 404s on the dashboard after it was created so I abandoned the task - I'll try again at some point"_

**Expected behavior**
Users should be taken to a metrics config screen when there is a human readable error and information on how to update the application to make the dashboard valid.

**Additional context**
https://twitter.com/reconbot/status/1586114999801741312?s=20&t=3h47b8qaigCs8vy54cPE1g

Related?
https://github.com/rilldata/rill-developer/issues/1133
https://github.com/rilldata/rill-developer/issues/1115
https://github.com/rilldata/rill-developer/issues/1117
https://github.com/rilldata/rill-developer/issues/1060

**Comment by hamilton** [comment]
Assigning to @djbarnwal

**Comment by djbarnwal** [comment]
@magorlick Is there a way I can reproduce the 404 errors in the dashboard?

**Comment by hamilton** [comment]
I think we did not quite scope this right on the product side. This bug is a regression; previously, we had for a certain number of cases, we redirected the user back to the metrics config so they could fix the app-level issue. I think those error cases are (1) the model is invalid in some way or (2) the model / source does not exist. For other errors, we should be showing a 404 page with more descriptive information; I would opt for us to figure out how to show something in the window itself w/ 404 styling.

For instance,
- if I have a dashboard link that doesn't show a dashboard, we should show a 404.
- if I delete the source that powers a dashboard, however, it should go back to the metrics page and show the default error.
